app/src/Middleware/TrailingSlashMiddleware.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Middleware;
+
+use Slim\Middleware;
+
+/**
+ * Redirect trailing slashes
+ */
+class TrailingSlashMiddleware extends Middleware
+{
+    /**
+     * @var bool
+     */
+    private $addSlash;
+
+    /**
+     * Configure whether add or remove the slash.
+     *
+     * @param bool $addSlash
+     */
+    public function __construct($addSlash = false)
+    {
+        $this->addSlash = $addSlash;
+    }
+
+    /**
+     * @return void
+     */
+    public function call()
+    {
+        $request = $this->app->request;
```

Comment:
I added a PR to your branch. Updating to $this->app->request() allows that method to be mocked and tests to work.